matrix.h

```diff
@@ -72,6 +133,51 @@ class CMatrix
         m_ppMatrix = nullptr;
         m_rows = m_cols = 0;
     }
+
+    Node* operator[](size_t row)
+    {
+        if(row < m_rows){
+            return m_ppMatrix[row]; //a
+        }
+        throw std::invalid_argument( "row can't be greater then m_rows." );
+    }
+
+    value_type &operator()(size_t row, size_t col)
+    {
+        if(row < m_rows && col < m_cols){
+            return m_ppMatrix[row][col].getDataRef(); //b
```

Comment:
explore how to create operator value_type to class NodeMatrix
After that, you may remove: .getDataRef()